src/views/product-integrations-landing/components/searchable-integrations-list/index.tsx

```diff
@@ -193,6 +196,54 @@ export default function SearchableIntegrationsList({
 	const resultText = `${filteredIntegrations.length} ${
 		filteredIntegrations.length === 1 ? 'result found' : 'results found'
 	}`
+
+	/**
+	 * -- Why a loading skeleton? --
+	 *
+	 * This effect is necessary when users land on the page and query parameters
+	 * are in the URL. Since query parameters are only accessible client-side,
+	 * filters are not applied until the first client-side render.
+	 *
+	 * Because this page is pre-rendered server-side based on the props from
+	 * getStaticProps, a hydration mismatch will happen between the two renders.
+	 * Instead of rendering the unfiltered view server-side, we render a loading
+	 * skeleton until the first client-side render via `useEffect`.
+	 *
+	 * -- Why the setTimeout? --
+	 *
+	 * The purpose of using `setTimeout` to extend the loading skeleton rendering
+	 * is to prevent a flash between the "loading" and "loaded" states for users
+	 * with faster internet connections. Flashes have high risk for harming users,
+	 * and no one wants to wait too long for a page to load.
+	 *
+	 * To balance these two needs we show the loading skeleton slightly longer
+	 * than needed to prevent the flash, and we animate the loading skeleton to
+	 * keep users engaged while it's displayed.
+	 */
+	const [isLoadingQueryParams, setIsLoadingQueryParams] =
+		useState(hasQueryParams)
+	useEffect(() => {
+		if (!isLoadingQueryParams) {
+			return
+		}
+
+		if (
+			contextState.integrations.length !==
+			contextState.filteredIntegrations.length
+		) {
```

Comment:
This check will cause the loading to never resolve, in the event the query parameters filter logic does not filter out any integrations.  A concrete example is that Waypoint integrations are all built-ins right now:

https://dev-portal-git-ambintegrations-loading-skeleton-hashicorp.vercel.app/waypoint/integrations?flags=builtin